### PR TITLE
Add common error prefix

### LINKED ISSUE #253: Error strings should be prefixed
To ease debugging, all errors returned by the `opcua` module should be prefixed with `opcua: `

Example:

```golang
var ErrSlowConsumer = errors.New("opcua: slow consumer. messages dropped")
```

### PR DESCRIPTION
**Pull request by kung-foo**
Fixes #253

### CODE REVIEW

File: ua/node_id_test.go [hunk 1]
```diff
@@ -391,3 +392,13 @@ func TestSetNamespace(t *testing.T) {
 		})
 	}
 }
+
+func compareErrors(err1, err2 error) bool {
```
Comment (magiconair) [inline ua/node_id_test.go:396]: I think `reflect.DeepEqual` does exactly what `compareErrors` does. 

https://play.golang.org/p/iBDNJELixui

File: uapolicy/policyAes128Sha256RsaOaep.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/rsa"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/policyAes128Sha256RsaOaep.go]: maybe sort this properly

File: uapolicy/crypto_hmac.go [hunk 3]
```diff
@@ -3,7 +3,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/hmac"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/crypto_hmac.go:6]: sort pkg name

File: uapolicy/policyAes256Sha256RsaPss.go [hunk 4]
```diff
@@ -7,7 +7,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/rsa"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/policyAes256Sha256RsaPss.go]: sort

File: uapolicy/policyBasic128Rsa15.go [hunk 5]
```diff
@@ -7,7 +7,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/rsa"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/policyBasic128Rsa15.go]: sort

File: uapolicy/policyBasic256.go [hunk 6]
```diff
@@ -7,7 +7,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/rsa"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/policyBasic256.go]: sort

File: uapolicy/policyBasic256Sha256.go [hunk 7]
```diff
@@ -7,7 +7,7 @@ package uapolicy
 import (
 	"crypto"
 	"crypto/rsa"
-	"errors"
+	"github.com/gopcua/opcua/errors"
```
Comment (magiconair) [inline uapolicy/policyBasic256Sha256.go]: sort

File: uapolicy/securitypolicy.go [hunk 8]
```diff
@@ -9,10 +9,10 @@ package uapolicy
 
 import (
 	"crypto/rsa"
-	"errors"
-	"fmt"
 	"sort"
 
+	"github.com/gopcua/opcua/errors"
+
```
Comment (magiconair) [inline uapolicy/securitypolicy.go:15]: drop blank line

File: uasc/secure_channel.go [hunk 9]
```diff
@@ -124,7 +124,7 @@ func (s *SecureChannel) SendRequest(req ua.Request, authToken *ua.NodeID, h func
 func (s *SecureChannel) SendRequestWithTimeout(req ua.Request, authToken *ua.NodeID, timeout time.Duration, h func(interface{}) error) error {
```
Comment (magiconair) [inline uasc/secure_channel.go:124]: this belongs to a different patch

**Review by magiconair — COMMENTED** [review]
almost. Can you clean it up and then we merge?

File: ua/node_id_test.go (hunk 1 above)
Comment (kung-foo) [reply ua/node_id_test.go:396]: The underlying stack trace captures will be different:

https://play.golang.org/p/1WqB7u4BUf7

File: uasc/secure_channel.go (hunk 9 above)
Comment (kung-foo) [reply uasc/secure_channel.go:124]: I have no idea how this is contaminating my branch. I even redid this PR from scratch.

File: uapolicy/securitypolicy.go (hunk 8 above)
Comment (kung-foo) [reply uapolicy/securitypolicy.go:15]: are you using a different fmt tool? `go fmt` doesn't report any issues this or `uapolicy/crypto_hmac.go`

File: ua/node_id_test.go (hunk 1 above)
Comment (magiconair) [reply ua/node_id_test.go:396]: Damn. That's not nice since the error itself doesn't change. It doesn't really matter where it occurs.

File: uapolicy/securitypolicy.go (hunk 8 above)
Comment (magiconair) [reply uapolicy/securitypolicy.go:15]: No, I'm just a stickler for these things. I believe in consistency and that developers don't read code but pattern match it. So any deviation from the structure slows you down b/c you fall back to the slower "read code" mode.

File: ua/node_id_test.go (hunk 1 above)
Comment (magiconair) [reply ua/node_id_test.go:396]: I've added a comment to this function referring to your play.golang.org link. My guess is that this will bite us at some point when users expect `reflect.DeepEqual` to work.

Comment (magiconair) [reply ua/node_id_test.go:396]: Just got bitten by this in two ways. First, that `errors` now adds a prefix and then that the comparison did not work. I'll expose the compare function so that we can use that in tests but it would be nice if we could find a more generic solution.
